### DISCUSSION TITLE
Can't get agents installed

### DISCUSSION THREAD

**Discussion by cxgl** [amidaware/tacticalrmm#700] [General]
Hi all,

I can't seem to get agents to install no matter how I try. Dynamic EXE, powershell, even manually -- most times I end up with:

`tactical RMM: failed to download mesh agent: unexpected EOF `

Further, I seem to have gotten a winagent-v1.6.2, although when running said tacticalrmm.exe it now shows "additional tasks" and is no longer silent. Is there a way to downgrade what TRMM thinks is the latest agent?

Thank you all

**Comment by dinger1986** [comment]
Are your ports all opened? DNS working etc?

**Reply by cxgl** [reply]
I believe they are. I am running on a VM (not docker) behind an nginx server, and can get to https://mesh., api., and rmm. Port 4222 is also going to the VM.

**Reply by dinger1986** [reply]
Can you install an agent directly from mesh and see if it connects?

**Reply by cxgl** [reply]
To be clear, you'd like me to go to https://mesh.mydomain.com, and login? If I try the 'tactical' user I'd use to get into https://rmm.mydomain.com it fails with a bad username and password.

If I try the /agentinvite url from the initial install, it just says "not found".

Edit: I see -- it's the MeshCentral username/password.  I get "Unable to connect web socket, click to reconnect."

**Reply by dinger1986** [reply]
Need to use the details supplied during install for mesh and then try it - I see you found it.

Assistance can be provided on discord.

Please be aware proxy's are not supported - has to be something wrong there.

Check the documentation to check services are running etc but chances are it's the proxy at fault

**Reply by cxgl** [reply]
...so nginx is a sticking point. Got it. Will move to discord. Thanks, dinger1986!

**Reply by wh1te909** [reply]
sounds like install didn't work properly first time. no point in trying to fix it. just wipe your vps, reinstall and save the output of the install script and send it to either dinger or I on discord
